ipld/retriever.go

```diff
@@ -310,14 +292,17 @@ func (rs *retrievalSession) doRequest(ctx context.Context, q *quadrant) {
 				if rs.isReconstructed() {
 					return
 				}
-				rs.square[idx] = share
+				if rs.square.GetCell(uint(x), uint(y)) != nil {
```

Comment:
Is it possible that the caller can call `GetCell` with an index that does not exist inside the square, causing the short circuit and then subsequently try to set the cell of the non-existent index, causing a panic?

Reply:
No, it's not possible. The boundaries of x and y are defined by sqaureSize and quadrants

Reply:
This should be fuzzed eventually IMO.

Reply:
AFAIK, fuzzing helps with the testing of logic with arbitrary bytes input. In this case, its only about two numbers